src/charm.py

```diff
@@ -188,6 +178,10 @@ def __init__(self, *args):
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
         self.framework.observe(self.on.config_changed, self._configure_sdcore_nms)
+        self.framework.observe(
+            self.on["nms"].pebble_custom_notice, self._sync_network_config
```

Comment:
Can we call self._configure_sdcore_nms here ? self._sync_network_config is already called under centralized configure method. I think, the result does not change.

self.framework.observe(self.on["nms"].pebble_custom_notice, self._configure_sdcore_nms)

Reply:
That's actually intentional. I think it doesn't make sense to go through all the logic in the main callback in this case. gNB configuration change doesn't affect the NMS, which in this case is just a middle man. 
`self._sync_network_config` called in the main callback covers some edge cases (one such case is mentioned in the thread opened by @patriciareinoso ).